src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+export enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+export interface Figure {
+  shape: Shape;
+  color: string;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const values: number[] = [a, b, c].sort((x: number, y: number) => x - y);
+
+    if (values[0] + values[1] <= values[2]) {
+      throw new Error('Values are incorrect');
+    }
+
+    if (values[0] <= 0 || values[1] <= 0 || values[2] <= 0) {
+      throw new Error('Value is less than or equal to 0');
```

Comment:
The error message should be corrected to 'Values are less than or equal to 0' for grammatical accuracy.